org.eclipse.m2e.editor/src/org/eclipse/m2e/editor/pom/OverviewPage.java

```diff
@@ -342,13 +365,17 @@ private void createArtifactSection(FormToolkit toolkit, Composite composite, Wid
     Label packagingLabel = toolkit.createLabel(artifactComposite, Messages.OverviewPage_lblPackaging, SWT.NONE);
 
     artifactPackagingCombo = new CCombo(artifactComposite, SWT.FLAT);
+    packagingTypes = new LinkedHashSet<>();
+
+    packagingTypes.add("jar"); //$NON-NLS-1$
+    packagingTypes.add("war"); //$NON-NLS-1$
+    packagingTypes.add("ejb"); //MNGECLIPSE-688 : add EAR & EJB Support //$NON-NLS-1$
+    packagingTypes.add("ear"); //$NON-NLS-1$
+    packagingTypes.add("pom"); //$NON-NLS-1$
+    packagingTypes.add("maven-plugin"); //$NON-NLS-1$
+    updateAvailablePackagingTypes(); // dynamically load available packging types from build plugins
```

Comment:
I think you could simply place the code from updateAvailablePackagingTypes() right here and safe the field `packagingTypes` then and use a local variable instead.

Reply:
Have this done but just waiting for final review from @mickaelistria so I can just push all changes at once.

Reply:
Just squash your commits and push them here, so we can see the current status for this

Reply:
+1 with @laeubi here, please push your progress without waiting for anyone. FWIW, I'm fine with this new patch, and would also like to see the changes suggested by @laeubi implemented. After that, I think it's find to merge as it's an isolated change bringing a great feature.

Reply:
Alright, pushed my change. Everything should be here now.